task/mgmt-gateway/src/update_buffer.rs

```diff
@@ -62,29 +67,53 @@ impl<T, const BLOCK_SIZE: usize> UpdateBuffer<T, BLOCK_SIZE> {
         }
     }
 
+    pub fn ensure_matching_stream_id(
+        &self,
+        stream_id: u64,
+    ) -> Result<(), ResponseError> {
+        match self.current_update_stream_id {
+            None => Err(ResponseError::UpdateNotPrepared),
```

Comment:
Nit: is this the right error code?

Earlier in the PR, `UpdateNotPrepared` seems to mean "you sent a chunk while the update while still in the `UpdatePrepare` state", rather than "there is no current update happening at all".